### ISSUE TITLE
TIDAL - Multiple Artists Scrobbling

### ISSUE THREAD

**Issue by forenta** [web-scrobbler/web-scrobbler#3686]
**Describe the bug**
Scrobbling multiple artists from TIDAL ends up with something the artists coming out like this:

    Beyonce / Kendrick Lamar
    INTO BeyonceKendrick Lamar

**How to reproduce**
1. Sign in at Tidal
2. Enable Web Scrobbler
3. Open & play https://tidal.com/browse/track/289971161
4. Have a look at the Web Scrobbler

**Expected behavior**
Either
_Thundercat, Tame Impala_
or
_Thundercat & Tame Impala_
instead of
_ThundercatTame Impala_
as artist.

**Screenshots**
![grafik](https://user-images.githubusercontent.com/2451238/235321828-905f7d72-8b1d-4620-bb58-92bdf921490a.png)

**Environment (please complete the following information)**
-   OS: Microsoft Windows 11
-   Browser: Microsoft Edge 112.0.1722.64
-   Extension version: 2.82.1

**Additional context**
We´ve seen this issue before: https://github.com/web-scrobbler/web-scrobbler/issues/960

**Comment by inverse** [comment]
@jaccarmac you're a TIDAL user right? Would you be able to take a look at this one :pray:

**Comment by jaccarmac** [comment]
Yes. I've been playing with V3, which is pretty broken for TIDAL at the moment. Looks like the site may have changed along with the extension.

**Comment by jaccarmac** [comment]
The comma looks to be in `::before` now, which is... annoying. I'm not sure what the best way to deal with this is; I suppose we'll need to special-case it in any case.

**Comment by manu-cyber** [comment]
What would a special treatment for Tidal look like? I would really like to not having to add commas manually anymore O:-)

**Comment by jaccarmac** [comment]
I've been doing the busywork occasionally myself, but don't scrobble as much from the desktop, hence the slowness in fixing this. Excuse aside: The special case goes something like this (I think). The scrobbler's default behavior is to look for a DOM element or set of DOM elements, and pull the concatenated inner text from the found set. Since the commas are now implemented as CSS, they don't exist in said inner text. So what's needed is a slightly different function to do the digging, which will target each artist name element and splice them with `", "`.

As an aside, as part of TIDAL's ongoing changes, I started noticing artist name order inconsistency on the web today. I've seen it before, but only on mobile and only after pausing in the middle of a song queue for a while. [Example](https://listen.tidal.com/album/296733442): The correct order is "Fedez, Annalisa, Articolo 31" but sometimes Fedez and Articolo 31 will be swapped.

**Comment by manu-cyber** [comment]
I had a go at it this evening but my absolute lack of TypeScript knowledge held me back ^^

I added a metadata filter as described in the wiki [0] which I would then use to concatenate the artists using `Util.joinArtists()` (or just `join()` if `joinArtists()` is inappropriate here).  
The only thing I _think_ I’m missing is getting the array with the one or more artist elements inside the `artistSelector`. I have no idea how to get those, although I imagine it should be pretty simple. I looked in a few random other connectors but couldn’t find what I was looking for.

Do you maybe have a hint for me here? :)
I pushed my work in progress [1] if you’d like to have a look.

[0] https://github.com/web-scrobbler/web-scrobbler/wiki/Connectors-development#metadata-filters
[1] https://github.com/web-scrobbler/web-scrobbler/compare/master...manu-cyber:web-scrobbler:fix/tidal-artist-split
